libs/vkd3d/state.c

```diff
@@ -2802,6 +2806,23 @@ static HRESULT d3d12_pipeline_state_init_graphics(struct d3d12_pipeline_state *s
     graphics->vertex_buffer_mask = mask;
     vkd3d_shader_free_shader_signature(&input_signature);
 
+    for (i = 0; i < D3D12_IA_VERTEX_INPUT_RESOURCE_SLOT_COUNT; i++)
+    {
+        if (mask & (1u << i))
+        {
+            /* In D3D12, VBO strides must be aligned to min(4, max(sizeof(format))) for a given input slot.
+             * Native drivers and Windows Vulkan drivers happen to be robust against buggy applications
+             * which use unaligned formats, but Vulkan drivers are not always robust here (they don't have to be).
+             * AMD Windows D3D12/Vulkan drivers return the wrong result for unaligned VBO strides,
+             * but it doesn't crash at least!
+             * RDNA will hang if we emit tbuffer_loads which don't conform to D3D12 rules. */
+
+            /* All VBO types in D3D12 are POT sized, so this works. */
```

Comment:
`R32G32B32_FLOAT` is not POT (although four bytes per channel, so I guess it doesn't matter here?)

Reply:
Clarified in comment.